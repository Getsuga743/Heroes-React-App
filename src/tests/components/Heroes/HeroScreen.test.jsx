import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import HeroeScreen from '../../../components/Heroes/HeroScreen';

describe('Pruebas en <HeroScreen />', () => {
  const historyMock = {
    length: 10,
    push: jest.fn(),
    goBack: jest.fn(),
  };

  const wrapper = mount(
    <MemoryRouter initialEntries={['/heroe']}>
      <HeroeScreen history={historyMock} />
    </MemoryRouter>,
  );

  test('debe de mostrarse el componente redirect si no hay argumentos en la url', () => {
    expect(wrapper.find('Redirect').exists()).toBe(true);
  });
  test('debe de mostrar un hero si el parametro existe y se encuentra', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/heroe/marvel-spider']}>
        <Route path="/heroe/:heroeId" component={HeroeScreen} />
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.row').exists()).toBe(true);
  });
  test('debe de regresar a la pantalla anterior con PUSH', () => {
    const history = {
      length: 1,
      push: jest.fn(),
      goBack: jest.fn(),
    };
    const wrapper = mount(
      <MemoryRouter initialEntries={['/heroe/marvel-spider']}>
        <Route
          path="/heroe/:heroeId"
          component={() => <HeroeScreen history={history} />}
        />
      </MemoryRouter>,
    );
    wrapper.find('button').prop('onClick')();
    expect(history.push).toHaveBeenCalledWith('/');
    expect(history.goBack).not.toHaveBeenCalledWith('/');
  });
  test('debe de regresar a la pantalla anterior GoBack', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/heroe/marvel-spider']}>
        <Route
          path="/heroe/:heroeId"
          component={() => <HeroeScreen history={historyMock} />}
        />
      </MemoryRouter>,
    );
    wrapper.find('button').prop('onClick')();
    expect(historyMock.push).not.toHaveBeenCalledWith('/');
    expect(historyMock.goBack).toHaveBeenCalled();
  });
  test('debe de llamar al redirect si el heroe no existe', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/heroe/marvsadsa']}>
        <Route
          path="/heroe/:heroeId"
          component={() => <HeroeScreen history={historyMock} />}
        />
      </MemoryRouter>,
    );
    expect(wrapper.text()).toBe('');
  });
});
