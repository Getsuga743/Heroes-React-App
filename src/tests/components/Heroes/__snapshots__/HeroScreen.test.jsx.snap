// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroScreen /> debe de mostrar un hero si el parametro existe y se encuentra 1`] = `
<div
  className="container mt-5 animate__animated animate__fadeIn"
>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <img
        alt="marvel-spider portrait"
        className="img-thumbnail"
        src="../assets/heroes/marvel-spider.jpg"
      />
    </div>
    <div
      className="col-8"
    >
      <h3>
        Spider Man
      </h3>
      <ul
        className="list-group-flush"
      >
        <li
          className="list-group-item"
        >
          <b>
            Alter ego: 
          </b>
          Peter Parker
        </li>
        <li
          className="list-group-item"
        >
          <b>
            Publisher: 
          </b>
          Marvel Comics
        </li>
        <li
          className="list-group-item"
        >
          <b>
            First Appearence: 
          </b>
          Amazing Fantasy #15
        </li>
      </ul>
      <h5>
        Characters
      </h5>
      <p>
        Peter Parker
      </p>
      <button
        className="btn btn-outline-info"
        onClick={[Function]}
        type="button"
      >
        Return
      </button>
    </div>
  </div>
</div>
`;
